priv/posts/security/20241212210148_epmd-public-exposure.md

```diff
@@ -0,0 +1,114 @@
+}
+---
+EPMD, essential for Erlang and RabbitMQ clustering, is often exposed online—posing hidden security risks and requiring quick mitigation steps.
+---
+
+The Erlang Port Mapper Daemon (EPMD) is a built-in component that facilitates clustering across Erlang-based applications (including RabbitMQ). While EPMD is essential in trusted network environments, it was never intended for exposure on the public internet. Recent scans have revealed over 85,000 instances of publicly exposed EPMD, with roughly half associated with RabbitMQ servers.
+
+If left unsecured, an exposed EPMD endpoint can potentially grant attackers a foothold, enabling them to join your cluster, run arbitrary code, and compromise your systems. Fortunately, mitigation steps are straightforward: disable EPMD if you’re not clustering, or restrict it behind a firewall and proper network configuration.
+
+## TL;DR
+
+**Check if you’re exposed:**
+- Run a port scan:  
+  ```bash
+  nmap -p 4369 -sV [Your Public IP]
+  ```  
+  If the output shows:  
+  ```
+  PORT     STATE SERVICE
+  4369/tcp open  epmd
+  ```
+  Your EPMD port is exposed.
+
+**For Erlang/Elixir users:**
+- Remove or avoid using `-name` or `-sname` if you don’t need clustering.
+- Bind distribution to a non-public interface with `vm.args`:
+  ```bash
+  -kernel inet_dist_use_interface '{127, 0, 0, 1}'
+  ```
+- Use a firewall or security group rules to restrict external access.
+
+**For RabbitMQ users:**
+- RabbitMQ runs on Erlang and uses EPMD behind the scenes. Even if you never interact directly with Erlang, you may still be affected.
+- Check the RabbitMQ networking documentation to learn how to identify exposed ports and close them:  
+  [https://www.rabbitmq.com/docs/networking#epmd](https://www.rabbitmq.com/docs/networking#epmd)  
+- The Erlang Ecosystem Foundation’s security hardening guide explains what EPMD is, which ports it uses, and how to close them down:  
+  [https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/distribution](https://erlef.github.io/security-wg/secure_coding_and_deployment_hardening/distribution)
+
+## What is EPMD and How Does it Work?
+
+Erlang was designed from the ground up for concurrency, distribution, and fault tolerance. To support clustering (distributed Erlang nodes connecting and communicating seamlessly), Erlang relies on EPMD. EPMD runs as a small daemon that keeps track of Erlang nodes and their associated ports. When a node starts with `-name` or `-sname`, it registers with EPMD on the host machine’s port 4369.
+
+In a controlled environment—such as an internal network—this works well. EPMD makes it easy for nodes to discover each other, connect, and form distributed clusters.
+
+**Key points:**
+- EPMD listens on port 4369 by default.
+- It’s assumed that EPMD is only reachable from within a trusted network.
+- Many Erlang/Elixir build tools and release pipelines (like Rebar3, Mix, and Distillery releases) enable EPMD by default.
```

Comment:
I'm not quite sure how Mix behaves with the `remote.vm.args` file.

https://github.com/elixir-lang/elixir/blob/fc2d8d94c0e828f01e0b112efeb0645f99f8d111/lib/mix/lib/mix/tasks/release.init.ex#L56

Please fact-check :)